server/src/main/java/io/spine/server/stand/UpdateHandler.java

```diff
@@ -25,100 +25,153 @@
 import io.spine.base.FieldPath;
 import io.spine.base.FieldPaths;
 import io.spine.client.CompositeFilter;
-import io.spine.client.CompositeFilter.CompositeOperator;
 import io.spine.client.Filter;
 import io.spine.client.IdFilter;
 import io.spine.client.Subscription;
+import io.spine.client.SubscriptionUpdate;
 import io.spine.client.Target;
 import io.spine.client.TargetFilters;
 import io.spine.protobuf.TypeConverter;
+import io.spine.server.stand.Stand.NotifySubscriptionAction;
 import io.spine.server.type.EventEnvelope;
 import io.spine.type.TypeUrl;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
 
+import java.util.Optional;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
 
+import static com.google.common.base.Preconditions.checkState;
 import static io.spine.base.FieldPaths.getValue;
 import static io.spine.server.storage.OperatorEvaluator.eval;
 import static io.spine.util.Exceptions.newIllegalArgumentException;
 
 /**
- * Matches the incoming event against a subscription criteria.
- *
- * <p>The class descendants decide on how to turn the event envelope into a set of data pieces
- * suitable for filtering.
+ * Tells which {@linkplain SubscriptionUpdate subscription updates} an {@code Event}s occurred
```

Comment:
Please review the grammar. Also, please consider simplifying this sentence. Maybe it's because of all the Javadoc tags, but I had to read it a couple of times to get the meaning.